00_setup.Rmd

```diff
@@ -248,11 +277,111 @@ if (refresh_data == 2) {
+    locs <-
+      locs %>%
+      distinct(.data$loc_code)
+    
+    try(db_drop_table(con, "#locs"),
+        silent = TRUE)
+    
+    
+    # locs_oud <-
+    #   copy_to(con,
+    #           locs) %>%
+    #   inner_join(tbl(con, "vwDimMeetpunt") %>%
+    #                select(loc_wid = .data$MeetpuntWID,
+    #                       loc_code = .data$MeetpuntCode),
+    #              .,
+    #              by = "loc_code")
+    locs <-
+      tbl(con, "vwDimMeetpunt") %>%
+                   select(loc_wid = .data$MeetpuntWID,
+                          loc_code = .data$MeetpuntCode)
+
+  }
+  
+  xg3 <-
+    tbl(con, "ssrs_Precalc") %>%
+    # left_join(tbl(con, "DimMetingType"),
+    #           by = "MetingTypeWID") %>%
+    select(loc_wid = .data$MeetpuntWID,
+           hydroyear = .data$HydroJaar,
+           # method_code = .data$MetingTypeCode,
+           # method_name = .data$MetingTypeNaam,
+           lg3_lcl = .data$GLG_2,
+           hg3_lcl = .data$GHG_2,
+           vg3_lcl = .data$GVG_2,
+           lg3_ost = .data$GLG_1,
+           hg3_ost = .data$GHG_1,
+           vg3_ost = .data$GVG_1
+    ) %>%
+    filter(.data$hydroyear >= startyear,
+           .data$hydroyear <= endyear) %>%
+    inner_join(locs %>%
+                 select(.data$loc_wid,
+                        .data$loc_code) %>%
+                 distinct,
+               .,
+               by = "loc_wid") %>%
+    select(-.data$loc_wid)
+  
+  xg3 <-
+    switch(vert_crs,
+           local = xg3 %>% select(-contains("ost")),
+           ostend = xg3 %>% select(-contains("lcl")),
+           both = xg3
+    ) %>%
+    arrange(.data$loc_code,
+            .data$hydroyear)
   
-  tubes_xg3 <- tubes_hab %>% 
-      get_xg3(watina, startyear = year(now()) - 18, endyear = 2016, vert_crs = "local",
-              truncated =  TRUE, collect = TRUE)
+  if (collect) {
+    xg3 <-
+      xg3 %>%
+      collect
+  }
+  
+  return(xg3)
   
+}  
+  
+  tubes_xg3_alles <- tubes_hab %>% 
+      get_xg3_temp(watina, startyear = year(now()) - 18, endyear = 2016, vert_crs = "local",
+              truncated =  TRUE, collect = TRUE)
+  tubes_xg3 <- tubes_xg3_alles  %>% 
+    semi_join(tubes_hab, by = "loc_code")  
+  
+#toevoegen van pb gebruikt voor grondwaterstandstandsindicator die gelegen zijn in een gw-afhankelijk type
+  tubes_xg3_vmm <- data.frame(loc_code = rep(tubes_hab %>% 
+                                       filter(area_code == "VMM") %>% 
+                                       distinct(loc_code) %>% 
+                                       pull(loc_code),times = 31), 
+                              hydroyear = seq(from = 2016 - 30, to = 2016),
+                              lg3_lcl = rep(-1, 31))
+  
+  tubes_xg3 <- bind_rows(tubes_xg3, tubes_xg3_vmm)  
   #oplossen van UTF-probleem: strings worden in inbo-SQL-databanken opgeslagen in UTF-16, terwijl hier gewerkt wordt met UTF-8. Dit geeft een probleem bij de kable-functie
```

Comment:
Good news.
I can't currently update watina: 'package ‘watina’ is not available (for R version 3.6.1)'

Reply:
OK, will have a look at this.

Reply:
Did you use `install.packages()` or `remotes::install_github()`? See [installation instructions](https://inbo.github.io/watina/#installing-testing-and-using-the-watina-package) of watina.

It seems like a warning from `install.packages()`.

Reply:
A later version is installed (not an easy operation: had to quit rstudio, new session wasn't enough).
Github (inbo/watina@42df02d). I think it's 0.3.0 (because the commit points to it, but RStudio overview  mentions still version 0.2.6.9000)

Reply:
You just need a regular install of the `master` to obtain 0.3.0; the inbo/watina@42df02d commit corresponds to 0.2.6 and does not contain the workaround of inbo/watina#56 . You can check which version you have with `packageVersion("watina")`. Or at the bottom of any rendered function documentation.

When RStudio is malbehaving, just make it shut up and/or do the maintenance steps in a pure R console :wink:

Reply:
Sorry Floris,
The session info was from an old cache. I had to renew it. Now it shows: Github (inbo/watina@8f21b66)